agent/http.go

```diff
@@ -88,64 +87,64 @@ type HTTPServer struct {
 	denylist *Denylist
 }
 
-type templatedFile struct {
+// bufferedFile implements os.File and allows us to modify a file from disk by
+// writing out the new version into a buffer and then serving file reads from
+// that. It assumes you are modifying a real file and presents the actual file's
+// info when queried.
+type bufferedFile struct {
```

Comment:
`http.go` is quite large and seems to be mostly the "framework" we use for the http API. It might be nice to move this type to a separate file at some point.